job/src/slipstream/job/executor.py

```diff
@@ -71,6 +72,10 @@ def _process_jobs(self, thread_name):
                     job.update_job(state='SUCCESS', return_code=return_code)
                     queue.consume()
                     self.logger.info(self._log_msg('Job "{}" finished'.format(job_uri), name=thread_name))
+            except NonexistentJobError as e:
+                self.logger.warning(self._log_msg('Job does not exist; removing from queue. Message: {}').format(e.reason))
+                queue.consume()
+                continue
             except:
```

Comment:
There is an open question on whether jobs should be removed for any fatal error.  The danger in doing that is that a server misconfiguration or backoff status code then might incorrectly clear the queue.